packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -1049,33 +1049,36 @@ class ProjectFileInvalidator {
     assert(packagesPath != null);
 
     if (lastCompiled == null) {
+      // Initial load.
       assert(urisToMonitor.isEmpty);
       return <Uri>[];
     }
 
-    final List<Uri> invalidatedFiles = <Uri>[];
-    int scanned = 0;
     final Stopwatch stopwatch = Stopwatch()..start();
-    for (Uri uri in urisToMonitor) {
-      if ((platform.isWindows && uri.path.contains(_pubCachePathWindows))
-          || uri.path.contains(_pubCachePathLinuxAndMac)) {
-        // Don't watch pub cache directories to speed things up a little.
-        continue;
-      }
+
+    final List<Uri> urisToScan = <Uri>[
+      // Don't watch pub cache directories to speed things up a little.
+      ...urisToMonitor.where((Uri uri) => !_isInPubCache(uri)),
+
+      // We need to check the .packages file too since it is not used in compilation.
+      fs.file(packagesPath).uri,
+    ];
+    final List<Uri> invalidatedFiles = <Uri>[];
+    for (final Uri uri in urisToScan) {
       final DateTime updatedAt = fs.statSync(
           uri.toFilePath(windows: platform.isWindows)).modified;
-      scanned++;
       if (updatedAt != null && updatedAt.isAfter(lastCompiled)) {
         invalidatedFiles.add(uri);
       }
     }
-    // We need to check the .packages file too since it is not used in compilation.
-    final DateTime packagesUpdatedAt = fs.statSync(packagesPath).modified;
-    if (packagesUpdatedAt != null && packagesUpdatedAt.isAfter(lastCompiled)) {
-      invalidatedFiles.add(fs.file(packagesPath).uri);
-      scanned++;
-    }
-    printTrace('Scanned through $scanned files in ${stopwatch.elapsedMilliseconds}ms');
+    printTrace(
+      'Scanned through ${urisToScan.length} files in '
+      '${stopwatch.elapsedMilliseconds}ms',
+    );
     return invalidatedFiles;
   }
+
+  static bool _isInPubCache(Uri uri) =>
```

Comment:
This is formatted kind of weird. See https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#align-expressions

Reply:
Did you mean `_isInPubCache`'s boolean expression?  Fixed.